python/ray/tune/tests/test_sample.py

```diff
@@ -193,6 +193,32 @@ def testQuantized(self):
         samples = tune.sample.Float(0, 33).quantized(3).sample(size=1000)
         self.assertTrue(all(0 <= s <= 33 for s in samples))
 
+    def testCategoricalSeedInTrainingLoop(self):
+        def train(config):
+            return 0
+
+        config = {
+            "integer": tune.randint(0, 100_000),
+            "choice": tune.choice(list(range(100_000)))
+        }
+
+        np.random.seed(1000)
+
+        out_1 = tune.run(train, config=config, num_samples=8, verbose=0)
+
+        integers_1 = [t.config["integer"] for t in out_1.trials]
+        choices_1 = [t.config["choice"] for t in out_1.trials]
+
+        np.random.seed(1000)
```

Comment:
just for my own understanding: this is technically not necessary, right? let's leave it in b/c I think it helps w/ code readability, but `tune.run` should never change np's random seed, correct?

Reply:
Here we are running two experiments. Running the first experiment will change numpy's random state by sampling configurations, and we reseed it to get the same results on the second run.